### PR TITLE
Feature/37/annotations

### PR DESCRIPTION
**Pull request by trevrosen**
## What
Support for the Annotations API

## Why
Completes API coverage for `1.0.0` by adding capabilities for Annotation Events, Streams, and Links.

### CODE REVIEW

File: annotations.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
+package appoptics
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type AnnotationStream struct {
```
Comment (seubert) [inline annotations.go]: Normally I'm not a huge stickler on this, but we should probably document everything that's exported for a nice godoc.

File: appoptics_test/server_mux_test.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ func teardown() {
 }
 
 func TestMain(m *testing.M) {
+	defer server.Close()
```
Comment (seubert) [inline appoptics_test/server_mux_test.go]: Can we remove the `teardown` function if we're doing this instead?

File: annotations.go (hunk 1 above)
Comment (seubert) [reply annotations.go]: (I meant to add since it's a public API/lib)

Comment (trevrosen) [reply annotations.go]: will do

File: appoptics_test/server_mux_test.go (hunk 2 above)
Comment (trevrosen) [reply appoptics_test/server_mux_test.go]: thought a bit more and just keeping `teardown`

**Pull request merged**